pages/docs/guides/message-validation.md

```diff
@@ -0,0 +1,118 @@
+Check out the Message validation in the runtime() tutorial if you want to get your hands dirty with message validation. 
+
+```mermaid
+graph TD
+    subgraph Producer
+    G[Message produced] -->|Message| B{Is Message valid?}
+    B -->|No| A[Logs]
+    end
+    B -->|Yes| C[Broker]
+    C -->|Message| E{Is Message valid?}
+    subgraph Consumer
+    E -->|No| F[Logs]
+    E -->|Yes| H[Message consumed]
+    end
+```
+
+### Validation in a gateway
+A gateway intercepts all incoming messages and routes them through the middleware and handler pipelines. The gateway sits between the producer and the broker. The messages are routed through the gateway first, and the gateway determines whether the message is valid. If the message is invalid, it displays an error and is not forwarded to the broker.
+
+An example implementation of message validation in a gateway is the [AsyncAPI gateway](https://github.com/asyncapi/event-gateway). It intercepts all incoming messages moving them into a pipeline of middlewares and handlers such as message validation. You can use a Kafka consumer/producer[(kcat)](https://github.com/edenhill/kcat), a broker, and a simple WebSocket to run the AsyncAPI gateway in your machine.
+Check out an [AsyncAPI file demo with Studio.](https://studio.asyncapi.com/?url=https://raw.githubusercontent.com/asyncapi/event-gateway/master/deployments/k8s/event-gateway-demo/event-gateway-demo.asyncapi.yaml)
+
+<Remember>
+Currently, only the Kafka protocol is supported.
+</Remember>
+
+```mermaid
+graph TD
+    PR[Producer]-- Message --- EG[AsyncAPI Event-Gateway]
+    EG -- Message --- EGV{Is Message valid?}
+    EGV -->|Yes| BR[Broker]
+    EGV --- |No| INV{Fail when invalid?}
+    INV -->|Yes| ERR[/Fail/] -- Produce request errored --> PR
+    INV -->|No| BR
+```
+The AsyncAPI document is important in this case because payload schemas are taken from it and messages as validated against it in your application.
+You can spin up the AsyncAPI gateway using an AsyncAPI file. All the messages are forwarded to a WebSocket endpoint; if the message/payload is invalid, it includes a validation error message.
+
+### Validation handled by the broker
+As producers and consumers do not communicate with each other directly, but rather information transfer happens via Kafka topic. At the same time, the consumer still needs to know the type of data the producer is sending. Imagine if the producer starts sending bad data to Kafka or if the data type of your data gets changed. We need a way to have a common data type that must be agreed upon.
+
+That’s where Schema Registry comes into the picture. It is an application that resides outside of your Kafka cluster and handles the distribution of schemas to the producer and consumer by storing a copy of schema in its local cache.
+
+```mermaid
```

Comment:
I'm not sure what text should be mapped to what diagram. So I see one diagram, then text that I think is about it, but then there is another diagram.

maybe the other diagram should be removed and the new one you added should be followed. I assume you wanted to replicate this https://github.com/asyncapi/website/pull/1005/files#r1045172876 in simplified way. Then let's go this direction, but at the moment it is not 100% accurate, I mean diagram do not match the behaviour you describe

Reply:
I'll try making a better one

Reply:
https://mermaid.ink/img/pako:eNqNkk9rAjEQxb_KMGdD77m19WJbtLh6C8iQRLu4m5T8EWTZ724SqkvTWnpLMr_3Mm-YAaVVGjnKjryft3Rw1AsDoFqnZWitgc1Tvpc6NPJD97TWh9YHd4YhVwDYibqogcPjydmHl2a1zO_jJHt3VkWpnb8KFAVKvC92i3mFP1vjY_9v_JX2RxqqxyzZXQW7xfLv-mq7mYAqJDBm4Sf_G2kZgynsreO7preg39Bp9Ou3u7_U-TJYBlGc6yoU3zpCQYsIZ5ja6KlVaRXKJASGBGqBPB0VuaNAYcbEUQy2ORuJPLioZxg_k6_-2hzke-q8Hi8878o2?type=png

I'm not able to get it right in mermaid

Reply:
@derberg ⬆️ 😬 help? hehe

Reply:
@starlightknown can you also add here a link to the mermaidJS editor with your markdown or put it in a code block here? I think that will help Lukasz help you faster too ✌🏽

Reply:
I'm not able to find the markdown now. Everytime I open it disappears unless it's the same link and now it's lost. I think we can proceed with the previous diagram and remove this new one as one diagram is enough to explain and that is well done already @alequetzalli what do you think?

Reply:
I already lost track of which diagram we're going with 😂 

sounds good, I will let Lukasz tell you if the diagram passes 👍